lightning/src/offers/offer.rs

```diff
@@ -1216,6 +1216,11 @@ impl TryFrom<FullOfferTlvStream> for OfferContents {
 			return Err(Bolt12SemanticError::MissingDescription);
 		}
 
+		if let Some(currency_bytes) = currency {
+			core::str::from_utf8(&currency_bytes)
+				.map_err(|_| Bolt12SemanticError::InvalidCurrencyCode)?;
```

Comment:
We should restrict this more such that each character `is_ascii_uppercase`. That way we won't get control characters, and the user can display the currency safely.